packages/fetchai/skills/erc1155_client/handlers.py

```diff
@@ -0,0 +1,237 @@
+            agents = oef_msg.agents
+            self._handle_search(agents)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_search(self, agents: List[str]) -> None:
+        """
+        Handle the search response.
+
+        :param agents: the agents returned by the search
+        :return: None
+        """
+        if len(agents) > 0:
+            self.context.logger.info(
+                "[{}]: found agents={}, stopping search.".format(
+                    self.context.agent_name, list(map(lambda x: x[-5:], agents))
+                )
+            )
+            strategy = cast(Strategy, self.context.strategy)
+            # stopping search
+            strategy.is_searching = False
+            # pick first agent found
+            opponent_addr = agents[0]
+
+            query = strategy.get_service_query()
+            self.context.logger.info(
+                "[{}]: sending CFP to agent={}".format(
+                    self.context.agent_name, opponent_addr[-5:]
+                )
+            )
+            cfp_msg = FIPAMessage(
+                message_id=FIPAMessage.STARTING_MESSAGE_ID,
+                dialogue_reference=(" ", " "),
+                performative=FIPAMessage.Performative.CFP,
+                target=FIPAMessage.STARTING_TARGET,
+                query=query,
+            )
+            self.context.outbox.put_message(
+                to=opponent_addr,
+                sender=self.context.agent_address,
+                protocol_id=FIPAMessage.protocol_id,
+                message=FIPASerializer().encode(cfp_msg),
+            )
+        else:
+            self.context.logger.info(
+                "[{}]: found no agents, continue searching.".format(
+                    self.context.agent_name
+                )
+            )
+
+
+class TransactionHandler(Handler):
+    """Implement the transaction handler."""
+
+    SUPPORTED_PROTOCOL = TransactionMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        tx_msg_response = cast(TransactionMessage, message)
+        self.context.logger.info(tx_msg_response)
+        if tx_msg_response.tx_id == "contract-sign-hash":
+            tx_signed = tx_msg_response.signed_payload.get("tx_signature")
+            new_message_id = 2
+            new_target = 1
+
+            inform_msg = FIPAMessage(
+                message_id=new_message_id,
+                dialogue_reference=("", ""),
+                target=new_target,
+                performative=FIPAMessage.Performative.INFORM,
+                info={"signature": tx_signed},
+            )
```

Comment:
Create the inform message. for the deployer once we signed the transaction. The deployer will send the transaction to the smart contract. And the smart contract will check the signatures. If the signatures match the trade will be completed.